cpp/include/raft/spatial/knn/knn.cuh

```diff
@@ -65,6 +66,60 @@ inline void knn_merge_parts(value_t* in_keys,
     in_keys, in_values, out_keys, out_values, n_samples, n_parts, k, stream, translations);
 }
 
+/**
+ * Performs a k-select across row partitioned index/distance
+ * matrices formatted like the following:
+ * row1: k0, k1, k2
```

Comment:
Does this mean that the matrices always have 3 columns?  If so, then would you consider expressing that at compile time in the `extents` types of the matrices?